packages/react/src/ButtonGroup/ButtonGroup.test.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {render, cleanup} from '@testing-library/react'
+import '@testing-library/jest-dom'
+
+import {ButtonGroup} from './ButtonGroup'
+import {Button} from '../Button'
+import {axe, toHaveNoViolations} from 'jest-axe'
+
+expect.extend(toHaveNoViolations)
+
+describe('ButtonGroup', () => {
+  afterEach(cleanup)
+
+  it('renders correctly into the document', () => {
+    const expectedClass = 'ButtonGroup'
+    const expectedTag = 'section'
+
+    const {getByTestId} = render(
+      <ButtonGroup data-testid="test">
+        <Button>Primary Action</Button>
+        <Button>Secondary Action</Button>
+      </ButtonGroup>
+    )
+    const buttonGroupEl = getByTestId('test')
+    expect(buttonGroupEl.tagName).toBe(expectedTag.toUpperCase())
+    expect(buttonGroupEl.classList).toContain(expectedClass)
+  })
+
+  it('renders buttons with the correct element type when buttonAs is set', () => {
+    const expectedTag = 'a'
+
+    const {getByTestId} = render(
+      <ButtonGroup buttonsAs={'a'}>
+        <Button data-testid="test">Primary Action</Button>
+        <Button>Secondary Action</Button>
+      </ButtonGroup>
+    )
+    const buttonEl = getByTestId('test')
+    expect(buttonEl.tagName).toBe(expectedTag.toUpperCase())
+  })
+
+  it('renders buttons with the correct size class when buttonSize is set', () => {
+    const expectedClass = 'Button--size-large'
+
+    const {getByTestId} = render(
+      <ButtonGroup buttonSize={'large'}>
+        <Button data-testid="test">Primary Action</Button>
+        <Button>Secondary Action</Button>
+      </ButtonGroup>
+    )
+    const buttonEl = getByTestId('test')
```

Comment:
Could getByRole('button') be more suitable here?

Reply:
What is the benefit of that over this approach?

Reply:
using test ids in testing library is a last resort matcher.. `byRole` is generally preferred approach. If for example Button changed its default tag from `button` to `a`, this test would also then fail. 

Also..

> "getByTestId: The user cannot see (or hear) these, so this is only recommended for cases where you can't match by role or text or it doesn't make sense (e.g. the text is dynamic)."  

https://testing-library.com/docs/queries/about/

Reply:
All fixed up this is good to know